File: src/main/scala/com/github/mauricio/async/db/postgresql/exceptions/MissingCredentialInformationException.scala
package com.github.mauricio.async.db.postgresql.exceptions

import com.github.mauricio.postgresql.messages.backend.AuthenticationResponseType

/**
 * User: mauricio
 * Date: 3/31/13
 * Time: 1:39 AM
 */
class MissingCredentialInformationException (
                                              val username : String,
                                              val password : Option[String],
                                              val authenticationResponseType : AuthenticationResponseType.AuthenticationResponseType )
  extends DatabaseException (
    "Username and password were required by auth type %s but are not available (username=[%s] password=[%s]".format(
      authenticationResponseType,
      username,
      password
    )
  )

File: postgresql-async/src/main/java/com/github/mauricio/async/db/postgresql/exceptions/MissingCredentialInformationException.kt
package com.github.mauricio.async.db.postgresql.exceptions

import com.github.jasync.sql.db.exceptions.DatabaseException
import com.github.mauricio.async.db.postgresql.messages.backend.AuthenticationResponseType

class MissingCredentialInformationException(
    val username: String,
    val password: String?,
    private val authenticationResponseType: AuthenticationResponseType)
  : DatabaseException(
    "Username and password were required by auth type %s but are not available (username=<%s> password=<%s>".format(
        authenticationResponseType,
        username,
        password
    )
)